elasticsearch5/src/main/java/org/vertexium/elasticsearch5/ElasticsearchSearchQueryBase.java

```diff
@@ -998,64 +1006,67 @@ protected QueryBuilder getFilterForComparePredicate(Compare compare, HasValueCon
-                        filters.add(QueryBuilders.rangeQuery(propertyName).gte(lower).lte(upper));
-                    } else {
-                        filters.add(QueryBuilders.termQuery(propertyName, value));
-                    }
-                    break;
-                case GREATER_THAN_EQUAL:
-                    if (has.value instanceof DateOnly) {
-                        value = ((DateTime) value).withTime(0, 0, 0, 0);
-                    }
-                    filters.add(QueryBuilders.rangeQuery(propertyName).gte(value));
-                    break;
-                case GREATER_THAN:
-                    if (has.value instanceof DateOnly) {
-                        value = ((DateTime) value).withTime(23, 59, 59, 999);
-                    }
-                    filters.add(QueryBuilders.rangeQuery(propertyName).gt(value));
-                    break;
-                case LESS_THAN_EQUAL:
-                    if (has.value instanceof DateOnly) {
-                        value = ((DateTime) value).withTime(23, 59, 59, 999);
-                    }
-                    filters.add(QueryBuilders.rangeQuery(propertyName).lte(value));
-                    break;
-                case LESS_THAN:
-                    if (has.value instanceof DateOnly) {
-                        value = ((DateTime) value).withTime(0, 0, 0, 0);
-                    }
-                    filters.add(QueryBuilders.rangeQuery(propertyName).lt(value));
-                    break;
-                case NOT_EQUAL:
-                    addNotFilter(filters, propertyName, value);
-                    break;
-                default:
-                    throw new VertexiumException("Unexpected Compare predicate " + has.predicate);
-            }
+            filters.add(getFilterForProperty(compare, has, propertyName, value));
         }
         return getSingleFilterOrOrTheFilters(filters, has);
     }
 
+    private QueryBuilder getFilterForProperty(Compare compare, HasValueContainer has, String propertyName, Object value) {
+        if (Element.ID_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.ELEMENT_ID_FIELD_NAME;
+        } else if (Edge.LABEL_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.EDGE_LABEL_FIELD_NAME;
+        } else if (Edge.OUT_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.OUT_VERTEX_ID_FIELD_NAME;
+        } else if (Edge.IN_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
+            propertyName = Elasticsearch5SearchIndex.IN_VERTEX_ID_FIELD_NAME;
+        } else if (Edge.IN_OR_OUT_VERTEX_ID_PROPERTY_NAME.equals(propertyName)) {
+            return QueryBuilders.boolQuery()
+                    .should(getFilterForProperty(compare, has, Edge.OUT_VERTEX_ID_PROPERTY_NAME, value))
+                    .should(getFilterForProperty(compare, has, Edge.IN_VERTEX_ID_PROPERTY_NAME, value));
```

Comment:
Since this is an `or` with possibly other criteria involved, should we be safe and set the `minimum_should_match` to `1`? This is from the docs for `should`:
```
If the bool query is in a query context and has a must or 
filter clause then a document will match the bool query even
if none of the should queries match.  In this case these 
clauses are only used to influence the score.
...
This behavior may be explicitly controlled by settings the minimum_should_match parameter.
```